src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/removeusercaseaccess/RemoveUserCaseAccessAboutToStartHandler.java

```diff
@@ -0,0 +1,96 @@
+import uk.gov.hmcts.reform.finrem.caseorchestration.client.DataStoreClient;
+import uk.gov.hmcts.reform.finrem.caseorchestration.client.IdamAuthApi;
+import uk.gov.hmcts.reform.finrem.caseorchestration.controllers.GenericAboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.FinremCallbackHandler;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.FinremCallbackRequest;
+import uk.gov.hmcts.reform.finrem.caseorchestration.mapper.FinremCaseDetailsMapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseAssignedUserRole;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseAssignedUserRolesResource;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.DynamicList;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.DynamicListElement;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseData;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+import java.util.List;
+
+@Service
+@Slf4j
+public class RemoveUserCaseAccessAboutToStartHandler extends FinremCallbackHandler {
+
+    // Separator element for DynamicListElement code
+    static final String CODE_VALUES_SEPARATOR = "~#~";
+
+    private final AuthTokenGenerator authTokenGenerator;
+    private final DataStoreClient dataStoreClient;
+    private final IdamAuthApi idamAuthApi;
+
+    public RemoveUserCaseAccessAboutToStartHandler(FinremCaseDetailsMapper finremCaseDetailsMapper,
+                                                   AuthTokenGenerator authTokenGenerator,
+                                                   DataStoreClient dataStoreClient, IdamAuthApi idamAuthApi) {
+        super(finremCaseDetailsMapper);
+        this.authTokenGenerator = authTokenGenerator;
+        this.dataStoreClient = dataStoreClient;
+        this.idamAuthApi = idamAuthApi;
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.ABOUT_TO_START.equals(callbackType)
+            && EventType.REMOVE_USER_CASE_ACCESS.equals(eventType)
+            && (CaseType.CONSENTED.equals(caseType) || CaseType.CONTESTED.equals(caseType));
+    }
+
+    @Override
+    public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(
+        FinremCallbackRequest finremCallbackRequest, String authToken) {
+
+        String caseId = String.valueOf(finremCallbackRequest.getCaseDetails().getId());
+        log.info("Remove User Case Access about to start callback for Case ID: {}", caseId);
+
+        List<CaseAssignedUserRole> caseAssignedUserRoles = getCaseAssignedUserRoles(authToken, caseId);
+        List<DynamicListElement> dynamicListElements = caseAssignedUserRoles.stream()
```

Comment:
Not sure if it's necessary but could we not put this into a separate method too? 

Something like:


`getUserCaseAccessList(authToken, caseId)`

Then we can call `getCaseAssignedUserRoles` within this method.

Reply:
Method refactored